connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/InboundConnectorDetails.java

```diff
@@ -22,51 +22,88 @@
 import java.util.Map;
 
 /** Group of inbound connector elements that share the same deduplication ID. */
-public record InboundConnectorDetails(
-    String type,
-    String tenantId,
-    String deduplicationId,
-    @JsonIgnore Map<String, String> rawPropertiesWithoutKeywords,
-    List<InboundConnectorElement> connectorElements) {
+public sealed interface InboundConnectorDetails {
```

Comment:
Non-blocking questions for my own understanding: 
- Why not use a `Factory`-type pattern here (like used elsewhere in the codebase)?
- Why use a sealed interface solely (it seems) to allow inheritors to selectively implement the interface? Couldn't the invalid implementation just implement the `properties()` as returning null, for example?
- Why mix validation logic with record constructors below instead of separating the validation concern into another class?

I could see good arguments for them, I'd just like to understand a bit better 😄

Reply:
* A factory pattern would perhaps be an overkill here (at least from my perspective). Unlike in other cases where we use it, we don't have any dependencies on other classes for instance creation (unless we decide to extract validation concern into another class)
* However, I can see how in the current form this code looks like an over-use of sealed interfaces 😄 I started designing this when the inheritors didn't have any common fields and an interface was mostly a marker, but then I moved a few fields to the parent space for convenience. I will try to improve the model here (and also extract the validation concern from the constructor, at least into a separate method).

Reply:
Here's what I ended up doing:
- I extracted the creation & validation logic into a static method in a separate class, `InboundConnectorDetailsUtil`
- I kept the sealed hierarchy because I liked the data consistency benefit that we get (example: I can't create an `InvalidDetails` object without an error, or a valid object without properties). Also, since we're checking the object type in several places, it's nice that we get compile-time validation in `switch` expression that we handled all possible types without adding a `default` branch to the switch expressions.

I hope this is more balanced in terms of readability, usability, and not-overengineering-things 😄